### PR TITLE
Guidelines for PR Related to Developer Doc

### PR DESCRIPTION
**Pull request by kmova**
Include some high-level guidelines to be followed when raising PRs for the developer documentation.

### CODE REVIEW

File: contribute/process/pr-workflow-developer-doc.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Contributing a Patch to Developer Documentation
```
Comment (mahebbar) [inline contribute/process/pr-workflow-developer-doc.md]: Patch?  Can it be Process instead?

File: contribute/process/pr-workflow-developer-doc.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+## Create your development branch. 
+
+- Fork the openebs repository and rebase with master
+- Create a new development branch in your forked repository with the following naming convention: *"task description-#issue"*
+
+  Example:
+     This change is being developed with the branch named : OpenEBS-DevDoc-PR-Workflow-#213
+
+## Commit your changes
+- Reference the issue along with a brief description in your commits
```
Comment (mahebbar) [inline contribute/process/pr-workflow-developer-doc.md]: Refer the issue number

File: contribute/process/pr-workflow-developer-doc.md (hunk 1 above)
Comment (kmova) [reply contribute/process/pr-workflow-developer-doc.md]: should it be - Contributing Process for Developer Documentation ?

File: contribute/process/pr-workflow-developer-doc.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+The developer documenation includes:
+- Architecture and Design documentation
+- Technical or Research Notes
+- FAQ
+- Process Documentation 
+- Generic / Miscellaneous Notes 
+
+Anything that will help other developers!
+
+## Submit an Issue describing your proposed change.
```
Comment (mahebbar) [inline contribute/process/pr-workflow-developer-doc.md]: reference here to creating an issue where labels for doc issues can be labelled  "documentation"

File: contribute/process/pr-workflow-developer-doc.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+Some general guidelines when submitting issues for developer documentation:
+- If the proposed change requires an update to the existing page, please provide a link to the page in the issue. 
+
+You can also help with some existing issues under this category available at [developer documentation issues list](https://github.com/openebs/openebs/labels/documentation%2Fdevel)
+
+## Create your development branch. 
+
+- Fork the openebs repository and rebase with master
```
Comment (mahebbar) [inline contribute/process/pr-workflow-developer-doc.md]: rebase? not familiar with the term...is it merge?

**Comment by AmitKumarDas** [comment]
I felt it will be helpful to have the steps with commands documented.
Even add the commands or steps when a developer has merge issues.
Same is the case for commit message template. Alternatively, if we can set a commit template to our projects that will be even better.

File: contribute/process/pr-workflow-developer-doc.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+## Commit your changes
+- Reference the issue along with a brief description in your commits 
+- The commit summary should be within 50 characters
+- A detailed commit message COULD be included explaining details in each file. 
+
+## Submit a Pull request. 
+- Rebase your development branch 
+- Submit the PR from the development branch to the openebs/openebs:master
+- Incorporate review comments if any in the development branch.
```
Comment (mahebbar) [inline contribute/process/pr-workflow-developer-doc.md]: , if any,

Comment (kmova) [reply contribute/process/pr-workflow-developer-doc.md]: done!

File: contribute/process/pr-workflow-developer-doc.md (hunk 4 above)
Comment (kmova) [reply contribute/process/pr-workflow-developer-doc.md]: modified the sentence a bit, however rebase should be also ok..

File: contribute/process/pr-workflow-developer-doc.md (hunk 2 above)
Comment (kmova) [reply contribute/process/pr-workflow-developer-doc.md]: done.

File: contribute/process/pr-workflow-developer-doc.md (hunk 1 above)
Comment (kmova) [reply contribute/process/pr-workflow-developer-doc.md]: Changed the wording a bit, please review if it is appropriate.

**Comment by kmova** [comment]
@prateekpandey14 @AmitKumarDas 
It is a good idea to include actual commands. Will work on it.

File: contribute/process/pr-workflow-developer-doc.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+## Setting up Development Environment
+```
+git clone https://github.com/$user/openebs.git
+cd openebs/
+git remote add upstream https://github.com/openebs/openebs.git
+git remote set-url --push upstream no_push
+git remote -v
+```
+
+## Synchronizing your local master with remote(or upstream) master
```
Comment (AmitKumarDas) [inline contribute/process/pr-workflow-developer-doc.md]: What is the purpose of this section "Synchronizing your local master with remote master"

Comment (kmova) [reply contribute/process/pr-workflow-developer-doc.md]: Before we create a new branch from an existing forked repository, we need to sync local master with remote master

Comment (AmitKumarDas) [reply contribute/process/pr-workflow-developer-doc.md]: Fine. Then the last section should rebase from local master ?
The second section need not have git push ?

Comment (kmova) [reply contribute/process/pr-workflow-developer-doc.md]: having this option helps a developer in working on multiple issues -- while waiting on the reviews..

Comment (AmitKumarDas) [reply contribute/process/pr-workflow-developer-doc.md]: Yup. This section should be there. However, what is the use of git push here ?

File: contribute/process/pr-workflow-developer-doc.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+git push origin master
+```
+
+## Synchronizing (or rebase) your local branch with remote(or upstream) master
+```
+git checkout <branch-name>
+git fetch upstream master
+git rebase upstream/master
+git status
+git push
```
Comment (AmitKumarDas) [inline contribute/process/pr-workflow-developer-doc.md]: Is git push == push upsteram master ?

File: contribute/process/pr-workflow-developer-doc.md (hunk 6 above)
Comment (kmova) [reply contribute/process/pr-workflow-developer-doc.md]: as part of the sync, if some changes were merged, it needs to merge them to github/user/openebs:master

File: contribute/process/pr-workflow-developer-doc.md (hunk 7 above)
Comment (kmova) [reply contribute/process/pr-workflow-developer-doc.md]: we shouldn't push anything to upstream master. only to the forked (orig) master

**Pull request merged**